### DISCUSSION TITLE
Problem printing special character '█'  aka '\u2588' or ASCII Character 219

### DISCUSSION THREAD

**Discussion by JDwyer0097** [pybricks/support#1820] [Q&A]
I'm using the block character to construct a visual map of a maze, but use of the character seems to be unreliable. In the image below, you can see the first image is drawn correctly. The second image is mostly correct except for the top right. The bigger the map the more it mucks up. 

Is it likely something I'm doing wrong or is it an issue with printing special characters from Pybricks? Any ideas of solutions?

![image](https://github.com/user-attachments/assets/0b084093-c8c5-419a-9b0e-b362994b202c)

Below is sample code that should demonstrate the error. Note if you make it a 3x3 maze it prints correctly. However if you make it a 20x20 maze it has errors.
```
from pybricks.hubs import PrimeHub
from pybricks.pupdevices import Motor, ColorSensor, UltrasonicSensor, ForceSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.robotics import DriveBase
from pybricks.tools import wait, StopWatch

hub = PrimeHub()

walls = '\u2588'
path = '.'

# Create initial Tile array with walls
Maze_Array = [[walls]*20 for i in range(20)]
Maze_Array[2][1] = 'S' # First Tile
Maze_Array[1][1] = path # First Wall

def print_array():
    print("Modified 2D array:")
    for row in Maze_Array:
        #print(*row, sep ='')
        print(''.join(row))

print_array()
```

**Comment by BertLindeman** [comment]
Hello,

This is probably fixed (but not published yet) by the fix for #1743

Bert

**Reply by BertLindeman** [reply]
This **is** a hack and takes a little bit of extra time, but most of the times prints OK.

By waiting before and after the print of an utf-8 character the transfer is split there, so the utf-8 character is received as a whole.

I did it like this:
```python
def print_array():
    print("Modified 2D array:")
    wait(20)
    for row in Maze_Array:
        for this_char in row:
            wait(20)
            print(this_char, end="")
            wait(20)
        print()

print_array()
```

After the fix has been published, all this waiting would be no longer needed.
But for the time being you might go on with your maze.

 
Bert

**Comment by JDwyer0097** [comment]
Hi Bert,
Thank you for the explanation and suggestion. Much appreciated and glad to hear it will be resolved soon.
